src/app/pages/reports-dashboard/components/report/report.component.ts

```diff
@@ -255,41 +262,62 @@ export class ReportComponent extends WidgetComponent implements OnInit, OnChange
 
   timeZoomReset(): void {
     this.zoomLevelIndex = this.zoomLevelMax;
-    const rrdOptions = this.convertTimespan(this.currentZoomLevel);
+    const rrdOptions = this.convertTimeSpan(this.currentZoomLevel);
     this.currentStartDate = rrdOptions.start;
     this.currentEndDate = rrdOptions.end;
     this.customZoom = false;
-
     const identifier = this.report.identifiers ? this.report.identifiers[0] : null;
     this.fetchReport$.next({ rrdOptions, identifier, report: this.report });
+    this.clearLastEndDateForCurrentZoomLevel();
+  }
+
+  clearLastEndDateForCurrentZoomLevel(): void {
+    Object.keys(this.lastEndDateForCurrentZoomLevel).forEach((key: ReportZoomLevel) => {
+      this.lastEndDateForCurrentZoomLevel[key] = null;
+    });
   }
 
   timeZoomIn(): void {
-    // more detail
     if (this.zoomLevelIndex === this.zoomLevelMax) {
       return;
     }
+
+    this.lastEndDateForCurrentZoomLevel[this.currentZoomLevel] = this.currentEndDate;
+
     this.zoomLevelIndex += 1;
-    const rrdOptions = this.convertTimespan(this.currentZoomLevel);
+    const rrdOptions = this.convertTimeSpan(
+      this.currentZoomLevel,
+      ReportStepDirection.Backward,
+      (this.currentStartDate + this.currentEndDate) / 2,
+    );
+
     this.currentStartDate = rrdOptions.start;
     this.currentEndDate = rrdOptions.end;
     this.customZoom = false;
-
     const identifier = this.report.identifiers ? this.report.identifiers[0] : null;
     this.fetchReport$.next({ rrdOptions, identifier, report: this.report });
   }
 
   timeZoomOut(): void {
-    // less detail
     if (this.zoomLevelIndex === this.zoomLevelMin) {
       return;
     }
     this.zoomLevelIndex -= 1;
-    const rrdOptions = this.convertTimespan(this.currentZoomLevel);
+
+    const halfPeriodMilliseconds = this.getHalfPeriodMilliseconds();
+
+    const rrdOptions = this.convertTimeSpan(
+      this.currentZoomLevel,
+      ReportStepDirection.Backward,
+      isToday(this.currentEndDate) ? this.currentEndDate
```

Comment:
Indentation is wrong, plus it's hard to follow. Best to change into `if` statements.